dapps/marketplace/test/index.js

```diff
@@ -47,12 +49,67 @@ const reset = async (sellerOgn, reload = false) => {
 const purchaseListing = async ({ buyer }) => {
   await pic(page, 'listing-detail')
   await changeAccount(page, buyer)
+  
+  // Note: Some prop causes the Purchase button to rerender after a few hundred milliseconds
+  // Pupeeter doesn't support click events if the button is not in the DOM
+  // Pupeeter throws if the element get unmounted when it tries to click
+  await waitFor(300)
```

Comment:
I don't think we should render the Purchase button at all unless it is ready... instead we should show a loading indicator or similar. Think we should try to avoid waiting in tests, as it generally makes them a) flaky and b) take longer to run

Reply:
Yes, this was quite time consuming to figure out. I changed the `Buy.js` component to render a disabled button with `Loading...` text when it is not ready. Still I found that component re-rendering twice after all the loading is complete (not instantly but with some time apart). 

Found messaging polling to be the one causing it. Fixed that too yet it failed randomly.

One strange thing that I noted is that the tests passed every time I had `SCREENSHOTS` env variable set. Figured out that it has to do with delay while taking screenshots. That's why I tried adding delay manually.

Will investigate it further and see how this can be fixed.